pkg/daemon/daemon.go

```diff
@@ -183,17 +183,25 @@ func (d *daemon) AddPeriodicUpdate() {
 					continue
 				}
 			} else {
-				allocatedGuid, err = d.guidPool.GenerateGUID("")
+				allocatedGuid, err = d.guidPool.GenerateGUID(string(pod.UID) + networkName)
```

Comment:
why Genereate GUID get args?

Reply:
Since GenerateGUID needs to call AllocateGUID, It passes the ID for who is allocating the generated guid.

Reply:
I think we need to change implementation in guid pool.
GenerateGUID  - should just GenerateGUID  and return it. 
and then you need to call AllocateGUID. 
also the AllocateGUID(podID, NetworkID) so you can save it in the map

Reply:
@moshe010  In this PR or another one?

Reply:
In this PR

Reply:
changed